### PR TITLE
Fields nested following

### PR DESCRIPTION
**Pull request by myovchev**
## Summary

Allow nested fields (array and object schema fields) to follow field values from the parent schema.

In order to be done, the nested fields should use syntax `<fieldToFollow`, which will result following of values of a field named `fieldToFollow` from the parent schema. `<<fieldToFollow` would follow a field from the grand parent, and so on.

Here is an example, that involves an imaginary field type `stringWithSuggestions` following a parent field:
```js
module.exports = {
  extend: '@apostrophecms/piece-type',
  options: {
    name: 'household',
    fields: {
      add: {
        petPreference: {
          type: 'select',
          label: 'Pet Preference',
          choices: [
            {
              value: 'cats',
              label: 'Cats'
            },
            {
              value: 'dogs',
              label: 'Dogs'
            }
          ],
          required: true
        },
        pets: {
          type: 'array',
          label: 'Pets',
          fields: {
            add: {
              name: {
                // Custom schema field like "string" but
                // offers the user suggestions based on "preference"
                type: 'stringWithSuggestions',
                label: 'Name',
                following: [ '<petPreference' ]
              }
            },
            inline: true
          }
        }
      }
    }
  } 
};
``` 

## What are the specific steps to test this change?

Open the Testbed app -> Testing -> Households Piece. Change "Pet Preferences" field and see the chosen value change throughout the nested piece schema.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [x] Related tests have been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## UNRELEASED
 
+### Adds
+
+* Allow sub-schema fields (array and object) to follow parent schema fields using the newly introduced `following: '<parentField'` syntax, where the starting `<` indicates the parent level. For example `<parentField` follows a field in the parent level, `<<grandParentField` follows a field in the grand parent level, etc. The change is fully backward compatible with the current syntax for following fields from the same schema level.
```
Comment (boutell) [inline CHANGELOG.md:7]: grandparent (no space)

Comment (boutell) [inline CHANGELOG.md:7]: grandparent level (no space)

File: modules/@apostrophecms/schema/index.js [hunk 2]
```diff
@@ -1261,6 +1264,28 @@ module.exports = {
...
       },
 
+      // If a field has a following property and a parent,
+      // hoist that property values to the parent,
+      // if they start with `<`.
+      hoistFollowingFieldsToParent(field, parent) {
+        if (!parent || !field.following) {
+          return;
+        }
+        const following = typeof field.following === 'string'
```
Comment (boutell) [inline modules/@apostrophecms/schema/index.js:1274]: Is this a thing? I'm pretty sure we don't support `following` as anything other than an array.

Comment (boutell) [reply modules/@apostrophecms/schema/index.js:1274]: But if there are counterexamples already in our code then that is how it is.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue [hunk 3]
```diff
@@ -342,12 +343,21 @@ export default {
     },
     getFollowingValues(item) {
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue:344]: * Shouldn't there be code sharing on this between object and array via the mixin?
* Is it tested for both inline and non-inline arrays?

**Review by boutell — CHANGES_REQUESTED** [review]
This is great, very close to what we discussed, a few concerns

File: modules/@apostrophecms/schema/index.js (hunk 2 above)
Comment (myovchev) [reply modules/@apostrophecms/schema/index.js:1274]: `const following = Array.isArray(field.following) ? field.following : [ field.following ];` is how it's done prior the change (code from the editor mixin).

File: modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue (hunk 3 above)
Comment (myovchev) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue:344]: Testbed does test both inline and non-inline arrays in different depts.
I've went for completely new mixin, as it makes sense only for fields having their own schema.

**Pull request merged**